src/entity.ts

```diff
@@ -52,27 +52,44 @@ export namespace entity {
    * provided.
    *
    * @class
-   * @param {number} value The double value.
+   * @param {number|string} value The double value.
    *
    * @example
    * const {Datastore} = require('@google-cloud/datastore');
    * const datastore = new Datastore();
    * const aDouble = datastore.double(7.3);
    */
-  export class Double {
+  export class Double extends Number {
+    private _entityPropertyName: string | undefined;
     type: string;
     value: number;
-    constructor(value: number) {
+    constructor(value: number | string | ValueProto) {
+      super(typeof value === 'object' ? value.doubleValue : value);
```

Comment:
What is the use case of constructing from a proto ?
If it needs to be supported int would probably make sense too.

Reply:
Int has the same support from proto, and it is part of the entityFromEntityProto path. This code mirrors the behavior of our Int class.